ui/app/pages/send/send-content/add-recipient/ens-input.component.js

```diff
@@ -53,16 +54,22 @@ export default class EnsInput extends Component {
     }
   }
 
-  // If an address is sent without a nickname, meaning not from ENS or from
-  // the user's own accounts, a default of a one-space string is used.
   componentDidUpdate (prevProps) {
     const {
       input,
     } = this.state
     const {
       network,
+      value,
     } = this.props
 
+    // Set the value of our input based on QR code provided by parent
+    if (input !== value && prevProps.value !== value) {
+      this.setState({ input: value })
```

Comment:
`setState` will not update the state synchronously, meaning the `input` value used below will be stale of both branches are executed (i.e. if the `value` is passed in here in the same render that the network changes).

Though, `this.onChange` will also update the state with this new input :thinking:. So the state will end up updated twice in this case, to the same value.

Perhaps, instead of calling `setState` here, you could pass `value` through to `this.onChange` as is done in the network change branch? i.e. put the `this.onChange({ target: { value: input } })` in a branch that's triggered by `value` being passed in _or_ by the network change. That would also trigger all of the ENS-related state updates, and call the `onChange` callback. I would assume those steps are still required in this case.